Assets/_Challenges/Scripts/AI Behaviours/Debug/LogBehaviour.cs

```diff
@@ -11,15 +12,46 @@ namespace MessyCoderCommunity.AI
     [CreateAssetMenu(fileName = "Log Behaviour", menuName = "Messy AI/Debug/Log")]
     public class LogBehaviour : GenericAiBehaviour<GameObject>
     {
-        [Tooltip("The message to display. This can include variables on the chalkboard using `{VARIABLE_NAME}`.")]
-        public string message = "{agent} says 'Hi'";
+        [SerializeField, Tooltip("The message to display. This can include variables on the chalkboard using `{VARIABLE_NAME}`.")]
+        string message = "{agent} says 'Hi'";
+
+        private Regex variableRegex;
+
+        public override void Initialize(GameObject agent, Chalkboard chalkboard)
+        {
+            base.Initialize(agent, chalkboard);
+
+            variableRegex = new Regex(@"\{([^}]*)\}");
+        }
 
         public override void Tick(Chalkboard chalkboard)
```

Comment:
How often does this actually get called? We may want to cache the results of the regex.

Reply:
I agree, and this is cached. Initialize is called once per behaviour. I have commented the base classes so depending on your editor this kind of info may bubble up if you hover over the override keyword.

Reply:
Just checked on this, turns out I have *not* commented the base class. I did, then I blew it away, then I reimplemented and did not comment. Adding to the PR.